libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragmentAbstract.java

```diff
@@ -219,6 +219,8 @@ public static MediaType getEditorMimeType(MediaFile mediaFile) {
         void onReplaceStoryEditedBlockActionSent();
         void onReplaceStoryEditedBlockActionReceived();
         boolean showPreview();
+        Map<String, Double> onRequestBlockTypeImpressions();
+        void onSetBlockTypeImpressions(Map<String, Double> impressions);
```

Comment:
Just FYI: Instead of redeclaring these methods, we could have this interface extend the `OnBlockTypeImpressionsEventListener` interface. Doing that would also allow us to just pass the `editorFragmentListener` as the implementation of the `OnBLockTypeImpressionsEventListener` interface directly instead of creating an anonymous implementation of the interface in `GutenbergEditorFragment::onCreateView`. I created a commit showing what this would look like [here](https://github.com/wordpress-mobile/WordPress-Android/commit/32265822f6a020e6b44f5c8fdbeb5267337e06eb).

I think this is a nice change, but no strong feelings. Definitely not a blocker, so only make this change if you like it better. 🙂

Reply:
Interesting. Thank you for sharing this approach. In particular, thank you for sharing the example diff, as it was difficult for me to follow the written description. 

As a novice Android developer, I crafted the current implementation following the preexisting patterns in the code base for similar features. However, I see how leveraging extensions could make this less verbose. 

That said, I would prefer to forego the change at this time, as it appears it would require additional changes to the Gutenberg Mobile Java interface as well. I'd prefer to keep that stable now that it has been approved and I hope to merge later today.